### DISCUSSION TITLE
How to get direct path to photos

### DISCUSSION THREAD

**Discussion by CornyVanBeck** [photoprism/photoprism#2345] [Q&A]
Hello everyone I have a Problem calling photos in Photoprism per API. I developed a small C# application and would like to get the photos via API call but somehow I'm not able to do so.

Expectation:
One can successfully log in via API and retrieve photos form a specified album and display this photos i a dedicated application.

What i got so far:
Call:
`POST https://example.domain.com/api/v1/session`

Header:
`X-Session-Id = 1234`

Body:
```
{
    "password":"XXX",
    "username": "admin"
}
```

Response:
```
{
  ...
  },
  "id": "8647a84d57d3ad1c9438aab3c7e2f0fdd7cc0947819bc1ee",
  "status": "ok"
}
```

Next I call for the photos:
Call:
`GET https://example.domain.com/api/v1/photos?count=60&offset=0&album=arbo6wq5khj8izic`

Header:
`X-Session-ID = 8647a84d57d3ad1c9438aab3c7e2f0fdd7cc0947819bc1ee`

Response:
```
[
  {
    "ID": "170",
    "UID": "prbu6hm1bpx0w5ej",
    "Type": "image",
    "TypeSrc": "",
    "TakenAt": "2022-05-13T19:36:57Z",
    "TakenAtLocal": "2022-05-13T19:36:57Z",
    "TakenSrc": "",
    "TimeZone": "UTC",
    "Path": "2022/05",
    "Name": "20220513_193636_47A4EB16",
    "OriginalName": "threema-20190501-201314-900dd95552888ab3",
    "Title": "Threema Dd",
    "Description": "",
    "Year": -1,
    "Month": -1,
    "Day": -1,
    "Country": "zz",
    "Stack": 0,
    "Favorite": false,
    "Private": false,
    "Iso": 0,
    "FocalLength": 0,
    "FNumber": 0,
    "Exposure": "",
    "Quality": 3,
    "Resolution": 1,
    "Color": 6,
    "Scan": false,
    "Panorama": false,
    "CameraID": 1,
    "CameraModel": "Unknown",
    "CameraMake": "",
    "LensID": 1,
    "LensModel": "Unknown",
    "LensMake": "",
    "Lat": 0,
    "Lng": 0,
    "CellID": "zz",
    "PlaceID": "zz",
    "PlaceSrc": "",
    "PlaceLabel": "Unknown",
    "PlaceCity": "Unknown",
    "PlaceState": "Unknown",
    "PlaceCountry": "zz",
    "InstanceID": "",
    "FileUID": "frbu6hm2u54emqz8",
    "FileRoot": "/",
    "FileName": "2022/05/20220513_193636_47A4EB16.jpg",
    "Hash": "2a5e8af76ab3121bba98750431311bcc9607e2c0",
    "Width": 768,
    "Height": 1024,
    "Portrait": true,
    "Merged": false,
    "CreatedAt": "2022-05-13T19:36:58.395830432Z",
    "UpdatedAt": "2022-05-13T19:44:51.43640223Z",
    "EditedAt": "2022-05-13T19:40:59Z",
    "CheckedAt": "2022-05-19T19:44:51Z",
    "DeletedAt": "0001-01-01T00:00:00Z",
    "Files": null
  },
...
]
```
Now i would like to build some link from the received information to call them in my C# gallery but i don't know how.

**Comment by CornyVanBeck** [comment]
Could someone please look into this?

**Reply by kvalev** [reply]
You need to add `merged=true` to the request, which will force the API to return the file paths.

**Reply by CornyVanBeck** [reply]
Hm, that does not give me the right information either. I now have the `Files` property filled but the information provided there is already present without `merged=true`. 
```
...
"DeletedAt": "0001-01-01T00:00:00Z",
    "Files": [
      {
        "UID": "frbu6hm2u54emqz8",
        "PhotoUID": "prbu6hm1bpx0w5ej",
        "Name": "2022/05/20220513_193636_47A4EB16.jpg",
        "Root": "/",
        "OriginalName": "threema-20190501-201314-900dd95552888ab3",
        "Hash": "2a5e8af76ab3121bba98750431311bcc9607e2c0",
        "Size": 130216,
        "Codec": "jpeg",
        "Type": "jpg",
        "Mime": "image/jpeg",
        "Primary": true,
        "Portrait": true,
        "Width": 768,
        "Height": 1024,
        "Orientation": 1,
        "AspectRatio": 0.75,
        "Colors": "661103216",
        "Luminance": "C79C0EB93",
        "Diff": 748,
        "Chroma": 9,
        "CreatedAt": "2022-05-13T19:36:58.395830432Z",
        "UpdatedAt": "2022-05-13T19:44:51.43640223Z",
        "Markers": []
      }
...
```

**Comment by CornyVanBeck** [comment]
Hello?

**Comment by CornyVanBeck** [comment]
I found something while searching for a solution. If I just use
`"FileName": "2022/05/20220513_193636_47A4EB16.jpg",` 
this property and attach that path to the URL following by `originals` then I get exactly what I want. Resulting into the following URL:
`https://example.domain.com/originals/2022/05/20220513_193636_47A4EB16.jpg`

Although I have to pass username and password as form data.

But that seems the wrong approach. Isn't there any other way to get this behavior?
